stl/inc/mutex

```diff
@@ -45,10 +45,6 @@ public:
     }
 #endif // ^^^ !defined(_DISABLE_CONSTEXPR_MUTEX_CONSTRUCTOR) ^^^
 
-    ~_Mutex_base() noexcept {
-        _Mtx_destroy_in_situ(_Mymtx());

```

Comment:
I could be convinced to retain this debug check for `~recursive_mutex`, which feels less performance-sensitive to me.

Reply:
I'm fine with dropping it.